src/core/analytics.js

```diff
@@ -392,16 +392,19 @@ class Analytics {
                 let transformedEvents;
                 if (transformationServerAccess) {
                   // filter the transformed event for that destination
-                  transformedEvents = transformedPayload.find(
+                  const filteredDestination = transformedPayload.find(
                     (e) => e.id === intg.destinationId,
-                  ).payload;
+                  );
+                  transformedEvents = filteredDestination ? filteredDestination.payload : undefined;
                 } else {
                   transformedEvents = transformedPayload;
                 }
-                // send transformed event to destination
-                transformedEvents.forEach((tEvent) => {
-                  this.sendDataToDestination(intg, { message: tEvent.event }, methodName);
-                });
+                if (transformedEvents) {
+                  // send transformed event to destination
+                  transformedEvents.forEach((tEvent) => {
+                    this.sendDataToDestination(intg, { message: tEvent.event }, methodName);
+                  });
+                }
```

Comment:
```suggestion
                // send transformed event to destination
                transformedEvents?.forEach((tEvent) => {
                  this.sendDataToDestination(intg, { message: tEvent.event }, methodName);
                });
```

Reply:
This is `analytics.js` file, AFAIR this doesn't work.

Reply:
Updated logic

Reply:
Is optional chaining not working?

Reply:
Done